src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -338,7 +335,19 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
 
 			// Set blob properties
 			if (blobProperties != null) {
-				blobProperties.configure(blob);
+				blobProperties.configure(blob, env);
+			}
+
+			// Set blob metadata
+			if (metadata != null) {
+				HashMap<String, String> metadataMap = blob.getMetadata();
+				for (AzureBlobMetadataPair pair : metadata) {
+					metadataMap.put(
+						Util.replaceMacro(pair.getKey(), env),
```

Comment:
Looks like Azure SDK will throw IllegalArugmentException for null, empty and white space metadata key/value. I've added some check.